pyterrier/transformer.py

```diff
@@ -54,8 +54,17 @@ class Transformer:
     """
         Base class for all transformers. Implements the various operators ``>>`` ``+`` ``*`` ``|`` ``&`` 
         as well as ``search()`` for executing a single query and ``compile()`` for rewriting complex pipelines into more simples ones.
+
+        Its expected that either ``.transform()`` or ``.transform_iter()`` be implemented by any class extending this - this rule
+        does not apply for indexers, which instead implement ``.index()``.
     """
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs) 
+        self._transform_implemented = type(self).transform != Transformer.transform
+        self._transform_iter_implemented = type(self).transform_iter != Transformer.transform_iter
+        # we cant test for either self._transform_implemented or  self._transform_iter_implemented here, due to indexers
```

Comment:
Could this be checked when the subclass is defined using `__init_subclass__`?

I think indexers should probably define the methods in their base class to throw an error.